### ISSUE TITLE
Why does the multi-scale test have no effect?

### ISSUE THREAD

**Issue by gjxgjxgjxgjx** [open-mmlab/mmdetection#3448]
The configuration file is as follows：

_base_ =[
    '../_base_/models/faster_rcnn_r50_fpn.py',
    '../_base_/datasets/x.py',
    '../_base_/schedules/schedule_1x.py', '../_base_/default_runtime.py'
]

model = dict(
    pretrained='open-mmlab://resnext101_32x4d',
    backbone=dict(
        type='ResNeXt',
        depth=101,
        groups=32,
        base_width=4,
        num_stages=4,
        out_indices=(0, 1, 2, 3),
        frozen_stages=1,
        norm_cfg=dict(type='BN', requires_grad=True),
        style='pytorch',
        dcn=dict(type='DCN', deformable_groups=1, fallback_on_stride=False),
        stage_with_dcn=(False, True, True, True)))

img_norm_cfg = dict(
    mean=[123.675, 116.28, 103.53], std=[58.395, 57.12, 57.375], to_rgb=True)

test_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(
        type='MultiScaleFlipAug',
        img_scale=[(1333, 640), (1333, 672), (1333, 704), (1333, 736),
                   (1333, 768), (1333, 800)],
        flip=True,
        transforms=[
            dict(type='Resize', keep_ratio=True),
            dict(type='RandomFlip'),
            dict(type='Normalize', **img_norm_cfg),
            dict(type='Pad', size_divisor=32),
            dict(type='ImageToTensor', keys=['img']),
            dict(type='Collect', keys=['img']),
        ])
]

data = dict(
            test=dict(pipeline=test_pipeline)
            )

optimizer = dict(type='SGD', lr=0.02, momentum=0.9, weight_decay=0.0001)
optimizer_config = dict(grad_clip=None)
# learning policy
lr_config = dict(
    policy='step',
    warmup='linear',
    warmup_iters=500,
    warmup_ratio=0.001,
    step=[14, 24, 30])
total_epochs = 36

data = dict(
    samples_per_gpu=2
)

Is there anything wrong?

**Comment by ZwwWayne** [comment]
What do you mean by saying `Why does the multi-scale test have no effect`? Is performance not as high as what you expected or what? Please be more specific, e.g., what is the baseline, what do you expect.

**Comment by yustaub** [comment]
what advices do you provide when choose the img_scale in test cfg? appreciate for your reply! @ZwwWayne

**Comment by gjxgjxgjxgjx** [comment]
@ZwwWayne 

I have found the problem, because I repeatedly defined data in the configuration file. In addition, I would like to ask if the results obtained by multi-scale training and testing are not good, is it the problem of my scale selection? Are there any recommended Settings？

**Comment by gjxgjxgjxgjx** [comment]
faster rcnn  coco datasets @ZwwWayne

**Comment by r03943158** [comment]
> @ZwwWayne
> 
> I have found the problem, because I repeatedly defined data in the configuration file. In addition, I would like to ask if the results obtained by multi-scale training and testing are not good, is it the problem of my scale selection? Are there any recommended Settings？

I have the same problem. The results of applying the sigle scale (1333,800) is as same the multi-scale [(1400,800),(1400,600),(1400,400)] setting on coco. Would you please tell me the change of the performance when you remove the repeatedly definition of the data set?

**Comment by ZwwWayne** [comment]
The setting is hyper-parameter need to be tuned for different models on different datasets. Usually, it depends on how the model is trained. For example, if the model is trained by multi-scale images, the testing scale could vary in a greater range. This also depends on the dataset and the model itself.
